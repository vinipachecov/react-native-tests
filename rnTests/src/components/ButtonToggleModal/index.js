import React from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';

import { theme } from '../../theme';

const ButtonToggleModal = ({ toggleModal }) => (
  <TouchableOpacity
    onPress={toggleModal}
    style={buttonStyles.container}
  >
    <Text style={buttonStyles.text}>
      Create new bill
    </Text>
  </TouchableOpacity>
);

ButtonToggleModal.propTypes = {
  toggleModal: PropTypes.func.isRequired,
};

const buttonStyles = StyleSheet.create({
  container: {
    backgroundColor: theme.colors.primary.light,
    padding: theme.spaces[1],
  },
  text: {
    color: theme.colors.white,
    fontSize: theme.fontSizes.headerTwo,
    fontWeight: 'bold',
  },
});

export default ButtonToggleModal;
