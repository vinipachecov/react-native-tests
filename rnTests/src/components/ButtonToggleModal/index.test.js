import React from 'react';
import { create, } from 'react-test-renderer';

import ButtonToggleModal from './';

//Faremos um test suite pois ele recebe outros parâmetros
describe('[Component] ButtonToggleModal', () => {

  const toggleModal = jest.fn();
  const tree = create(<ButtonToggleModal toggleModal={toggleModal} />);


  it('matches snapshot ', () => {    
    expect(tree.toJSON()).toMatchSnapshot();
  })
  

  it('Calls toggleModal when pressed', () => {
    // pegamos a raiz[já que não há filhos] e dentro das propriedades 
    // o que é passado para o componente
    tree.root.props.toggleModal();
    // * Não foi usado o getInstance pois como foi comentado ele não possui filhos


    expect(toggleModal).toHaveBeenCalled();
  })
  
})