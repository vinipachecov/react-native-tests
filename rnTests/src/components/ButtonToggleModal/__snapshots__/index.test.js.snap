// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] ButtonToggleModal matches snapshot  1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#5472d3",
      "opacity": 1,
      "padding": 4,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 24,
        "fontWeight": "bold",
      }
    }
  >
    Create new bill
  </Text>
</View>
`;
